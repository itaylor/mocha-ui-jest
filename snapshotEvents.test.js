const { snapshotSupport } = require('./index.js');

const { emitter, events } = snapshotSupport;

describe('snapshot events test', () => {
  afterEach(() => emitter.removeAllListeners());

  test('expect works while firing pass events', () => {
    let passCount = 0;
    let failCount = 0;
    let lastTestContext;
    emitter.on(events.SNAPSHOT_PASS, (testContext) => {
      lastTestContext = testContext;
      passCount++;
    });
    emitter.on(events.SNAPSHOT_FAIL, (testContext) => {
      lastTestContext = testContext;
      failCount++;
    });
    expect([{ that: 'this' }]).toMatchSnapshot();
    expect(passCount).toBe(1);
    expect(failCount).toBe(0);
    expect(lastTestContext).toMatchSnapshot();
  });

  test('expect works while firing fail events', () => {
    let passCount = 0;
    let failCount = 0;
    let lastTestContext;
    emitter.on(events.SNAPSHOT_PASS, (testContext) => {
      lastTestContext = testContext;
      passCount++;
    });
    emitter.on(events.SNAPSHOT_FAIL, (testContext) => {
      lastTestContext = testContext;
      failCount++;
    });
    expect(() => {
      expect([{ fails: Math.random() }]).toMatchSnapshot();
    }).toThrow();
    expect(passCount).toBe(0);
    expect(failCount).toBe(1);
    expect(lastTestContext).toMatchSnapshot();
  });
});
