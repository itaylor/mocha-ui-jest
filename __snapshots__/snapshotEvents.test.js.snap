// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect works while firing fail events 1`] = `
Array [
  Object {
    "fails": 0.48196021454099003,
  },
]
`;

exports[`expect works while firing fail events 2`] = `
Object {
  "file": "snapshotEvents.test.js",
  "fullTitle": "snapshot events test expect works while firing fail events",
  "snapshotFile": "__snapshots__/snapshotEvents.test.js.snap",
  "title": "expect works while firing fail events",
}
`;

exports[`expect works while firing pass events 1`] = `
Array [
  Object {
    "that": "this",
  },
]
`;

exports[`expect works while firing pass events 2`] = `
Object {
  "file": "snapshotEvents.test.js",
  "fullTitle": "snapshot events test expect works while firing pass events",
  "snapshotFile": "__snapshots__/snapshotEvents.test.js.snap",
  "title": "expect works while firing pass events",
}
`;
